homeassistant/components/asuswrt/bridge.py

```diff
@@ -236,38 +260,22 @@ async def _get_available_temperature_sensors(self) -> list[str]:
         availability = await self._api.async_find_temperature_commands()
         return [SENSORS_TEMPERATURES[i] for i in range(3) if availability[i]]
 
-    async def _get_bytes(self) -> dict[str, Any]:
+    @handle_errors_and_zip((IndexError, OSError, ValueError), SENSORS_BYTES)
+    async def _get_bytes(self) -> Any:
         """Fetch byte information from the router."""
-        try:
-            datas = await self._api.async_get_bytes_total()
-        except (IndexError, OSError, ValueError) as exc:
-            raise UpdateFailed(exc) from exc
-
-        return _get_dict(SENSORS_BYTES, datas)
+        return await self._api.async_get_bytes_total()
 
-    async def _get_rates(self) -> dict[str, Any]:
+    @handle_errors_and_zip((IndexError, OSError, ValueError), SENSORS_RATES)
+    async def _get_rates(self) -> Any:
         """Fetch rates information from the router."""
-        try:
-            rates = await self._api.async_get_current_transfer_rates()
-        except (IndexError, OSError, ValueError) as exc:
-            raise UpdateFailed(exc) from exc
-
-        return _get_dict(SENSORS_RATES, rates)
+        return await self._api.async_get_current_transfer_rates()
 
-    async def _get_load_avg(self) -> dict[str, Any]:
+    @handle_errors_and_zip((IndexError, OSError, ValueError), SENSORS_LOAD_AVG)
+    async def _get_load_avg(self) -> Any:
```

Comment:
Do we don't know any type hints for these functions?

Reply:
Same as previous comment